### DISCUSSION TITLE
How to use int in the name of the value?

### DISCUSSION THREAD

**Discussion by simalei** [nlohmann/json#3561] [Q&A] [answered]
How I should use int to create a name of the value in the example below?

    json j;
    for (size_t i = 0; i < example_vector.size(); i++) {
        j[/* here should be i*/] = {{"first value", example_vector[i].first}, {"second value", example_vector[i].second}};
        std::cout << j << std::endl;
    }

**Comment by falbrechtskirchinger** [comment, marked as answer]
Are you asking how to create a key of integral type? JSON requires keys to be strings. You could use `std::to_string(i)`.

Is there a particular reason why you don't want to use an array?

**Reply by simalei** [reply]
Thanks for the answer! Yeah, I have to use vector because I need it to be dynamic in size.

**Reply by falbrechtskirchinger** [reply]
> Thanks for the answer! Yeah, I have to use vector because I need it to be dynamic in size.

I think you misunderstood. I'm talking about a JSON array, i.e.
```json
[
   {"first value": "first", "second value": "second"},
   {"first value": "first", "second value": "second"},
   {"first value": "first", "second value": "second"}
]
```
instead of
```json
{
   "0": {"first value": "first", "second value": "second"},
   "1": {"first value": "first", "second value": "second"},
   "2": {"first value": "first", "second value": "second"}
}
```

JSON arrays **are** vectors.

**Reply by simalei** [reply]
Sorry for the misunderstanding. Yes, I have to use the second variant because I need to know the number of the string.
